_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -1066,6 +1075,95 @@ private function _process_onboarding( $data ) {
 			: join( ', ', $error );
 	}
 
+	/**
+	 * Retrieve the first active sidebar.
+	 *
+	 * @return array|WP_Error First active sidebar, error if none exists.
+	*/
+	static function get_first_sidebar() {
+		$active_sidebars = get_option( 'sidebars_widgets', array() );
+		unset( $active_sidebars[ 'wp_inactive_widgets' ], $active_sidebars[ 'array_version' ] );
+
+		if ( empty( $active_sidebars ) ) {
+			return new WP_Error( 'invalid_data', 'Invalid data.', 400 );
+		}
+		$active_sidebars_keys = array_keys( $active_sidebars );
+		return array_shift( $active_sidebars_keys );
+	}
+
+	/**
+	 * Add or update Business Address widget.
+	 *
+	 * @param array $business_address_content Value corresponding to the Business
+	 * Address as received from the settings endpoint.
+	 *
+	 * @return WP_Error|true True when data has been saved correctly,
+	 * error otherwise.
+	*/
+	static function handle_business_address( $business_address_content ) {
+		$first_sidebar = self::get_first_sidebar();
+
+		$widgets_module_active = Jetpack::is_module_active( 'widgets' );
+		if ( ! $widgets_module_active ) {
+			$widgets_module_active = Jetpack::activate_module( 'widgets', false, false );
+		}
+		if ( ! $widgets_module_active ) {
+			return new WP_Error( 'invalid_data', 'Failed to activate module.', 400 );
+		}
+
+		if ( $first_sidebar ) {
+			$title = $business_address_content['name'];
+			$address =
+				$business_address_content['city'] . ' ' .
+				$business_address_content['state'] . ' ' .
```

Comment:
A small suggestion here: since we need to sanitize these and clean out any empty ones, how about something like this (I know it might look a little repetitive, but it's explicit enough):

```
$title = isset( $address['name'] ) ? sanitize_text_field( $address['name'] ) : '';

$street = isset( $address['street'] ) ? sanitize_text_field( $address['street'] ) : '';
$city = isset( $address['city'] ) ? sanitize_text_field( $address['city'] ) : '';
$state = isset( $address['state'] ) ? sanitize_text_field( $address['state'] ) : '';
$zip = isset( $address['zip'] ) ? sanitize_text_field( $address['zip'] ) : '';

$full_address = implode( ' ', array_filter( array( $street, $city, $state, $zip ) ) );
```

Also, upon success, you can save `jpo_business_address`, like this (assuming you remove the other `update_option( 'jpo_business_address'` call):

```
$address_pieces = array(
	'name' => $title,
	'street' => $street,
	'city' => $city,
	'state' => $state,
	'zip' => $zip
);
update_option( 'jpo_business_address', $address_pieces );
```

It might look a little verbose, but it's essential that we take advantage of sanitization before saving both in the widget and in the option.

Reply:
Yes, previous version was missing filtering empty info and sanitization. Good points 👍